home/pi/auto_run.sh

```diff
@@ -548,6 +566,85 @@ then
     rm ~/first_run
 fi
 
+# Matrix Voice Hat Setup
+if [ -f ~/setup_matrix ]
+then
+    if [ ! -f matrix_setup_state.txt ]
+    then
+        echo ""
+        echo "========================================================================="
+        echo "Setting up Matrix Voice Hat. This will install the matrixio-kernel-modules and pulseaudio"
+        echo "This will require 3 reboots"
+        echo "Press any key to continue..."
+        read -N1 -s anykey
+    else
+        echo "Press any key to continue setting up Matrix Voice HAT"
+        read -N1 -s anykey
+    fi
+
+    if [ ! -f matrix_setup_state.txt ]
+    then
+        echo "Add Matrix repo and key and Update and install packages"
+        # add repo
+        curl https://apt.matrix.one/doc/apt-key.gpg | sudo apt-key add -
+        echo "deb https://apt.matrix.one/raspbian $(lsb_release -sc) main" | sudo tee /etc/apt/sources.list.d/matrixlabs.list
+        sudo apt-get update -y
+        sudo apt-get upgrade -y
+
+        echo "stage-1" > matrix_setup_state.txt
+        echo "Needs to reboot in case of kernel updates"
+        read -p "Press enter to continue reboot"
+        sudo reboot
+    else
+        matrix_setup_state=$( cat matrix_setup_state.txt)
+    fi
+
+    if [ $matrix_setup_state == "stage-1" ]
+    then
+        echo "installing matcixio-kernel-modules"
+        sudo apt install matrixio-kernel-modules -y
+
+        echo "installing pulseaudio"
+        sudo apt-get install pulseaudio -y
+        
+        echo "Need to reboot after installing matcixio-kernel-modules and pulseaudio"
+        read -p "Press enter to continue reboot"
+        echo "stage-2" > matrix_setup_state.txt
+        sudo reboot
+    fi
+
+    if [ $matrix_setup_state == "stage-2" ]
+    then
+        echo "setting Matrix as standard microphone..."
+        echo "========================================================================="
+        pactl list sources short
+        sleep 5
+        pulseaudio -k
+        pactl set-default-source 2
+        pulseaudio --start
+        amixer
+        amixer set Master 99%
+        amixer
+        sleep 2
+        amixer
+
+        mycroft-mic-test
+        
+        read -p "You should have heard the recording playback. Press enter to continue"
+
+        echo "========================================================================="
+        echo "updating the python virtual environment"
+        bash mycroft-core/dev_setup.sh
+
+        echo "stage-3" > matrix_setup_state.txt
+        read -p "Press enter to continue reboot! this should be the last one"
```

Comment:
read -p "Your Matrix microphone is now setup!  Press enter to perform the final reboot and start Mycroft."